pkg/loadbalancer/smooth_weighted_round_robin_test.go

```diff
@@ -0,0 +1,310 @@
+			Expect(results[s.name]).Should(Or(Equal(math.Ceil(expectedWeight)), Equal(math.Floor(expectedWeight))))
+		}
+	}
+
+	BeforeEach(func() {
+		lb = NewSmoothWeightedRR().(*smoothWeightedRR)
+		smoothTestingServers = []server{
+			{name: "server1", weight: 5},
+			{name: "server2", weight: 1},
+			{name: "server3", weight: 1},
+		}
+		rand.Seed(time.Now().UnixNano())
+		servers = []server{
+			{name: "server1", weight: randFloat()},
+			{name: "server2", weight: randFloat()},
+			{name: "server3", weight: randFloat()},
+		}
+
+		roundRobinServers = []server{
+			{name: "server1", weight: math.SmallestNonzeroFloat64},
+			{name: "server2", weight: math.SmallestNonzeroFloat64},
+			{name: "server3", weight: math.SmallestNonzeroFloat64},
+		}
+
+		rand.Shuffle(len(servers), func(i, j int) { servers[i], servers[j] = servers[j], servers[i] })
+	})
+
+	When("lb is created", func() {
+		It("its state is new", func() {
+			for _, l := range []*smoothWeightedRR{lb} {
+				validateEmptyLBState(l)
+			}
+		})
+	})
+
+	When("removing all items", func() {
+		It("should return an empty map", func() {
+			for _, l := range []*smoothWeightedRR{lb} {
+				addAllServers(l)
+				validateAllServersAdded(l)
+				l.RemoveAll()
+				validateEmptyLBState(l)
+			}
+		})
+	})
+
+	When("a single item is added", func() {
+		It("should fail on adding nil interface", func() {
+			for _, l := range []*smoothWeightedRR{lb} {
+				err := l.Add(nil, 100)
+				Expect(err).ToNot(BeNil())
+				validateEmptyLBState(l)
+			}
+		})
+
+		It("should fail on adding negative weight", func() {
+			for _, l := range []*smoothWeightedRR{lb} {
+				err := l.Add(servers[0], -100)
+				Expect(err).ToNot(BeNil())
+				validateEmptyLBState(l)
+			}
+		})
+
+		It("should return contain it", func() {
+			for _, l := range []*smoothWeightedRR{lb} {
+				_ = addServer(0, l)
+				validateServerAdded(0, l)
+			}
+		})
+
+		It("should return it all the time", func() {
+			for _, l := range []*smoothWeightedRR{lb} {
+				server := addServer(0, l)
+				for i := 0; i < 10; i++ {
+					Expect(l.Next()).To(Equal(server.name))
+				}
+			}
+		})
+
+		It("should not allow to add it again", func() {
+			for _, l := range []*smoothWeightedRR{lb} {
+				server := addServer(0, l)
+				err := l.Add(server.name, server.weight)
+				Expect(err).ToNot(BeNil())
+				Expect(l.ItemsCount()).To(Equal(1))
+			}
+		})
+	})
+
+	When("load balancing with all weights equal 0 - i.e Round Robin", func() {
+		It("should balance equaly", func() {
+			for _, l := range []*smoothWeightedRR{lb} {
+				var rrs = make([]server, len(roundRobinServers))
+				_ = copy(rrs, roundRobinServers)
+				for _, s := range rrs {
+					err := l.Add(s.name, s.weight)
+					Expect(err).To(BeNil())
+				}
+
+				for i := 0; i < 100; i++ {
+					for _, s := range rrs {
+						Expect(l.Next().(string)).To(Equal(s.name))
+					}
+				}
+			}
+		})
+	})
+
+	When("load balancing", func() {
+		It("should balance between the servers by weight", func() {
```

Comment:
Can you clarify this case further? Like have the `When` describe the weights being used?

```suggestion
	When("the items are weighted  by ...", func() {
		It("should correctly balance between them", func() {
```